comp/remote-config/rcclient/rcclientimpl/rcclient.go

```diff
@@ -297,6 +301,14 @@ func (rc rcClient) agentConfigUpdateCallback(updates map[string]state.RawConfig,
 		err = rc.settingsComponent.SetRuntimeSetting("log_level", mergedConfig.LogLevel, model.SourceRC)
 	}
 
+	switch streamLogsSource {
+	case model.SourceRC:
+		if *mergedConfig.EnableStreamLogs {
+			err = rc.settingsComponent.SetRuntimeSetting("enable_stream_logs", *mergedConfig.EnableStreamLogs, model.SourceRC)
+		}
+	default:
+		err = rc.settingsComponent.SetRuntimeSetting("enable_stream_logs", false, model.SourceRC)
+	}
```

Comment:
Not sure I fully understand this logic. If I'm reading it right:
- if the config source for stream logs is RC, set the runtime setting to whatever is in the configuration payload from RC
- if the config source is _not_ RC, disable the runtime setting.

If that's the case, it doesn't seem like anything can ever set the runtime setting to be sourced from RC, right? My intuition is that handling for this config value should be similar to the handling of the `LogLevel` attribute, to maintain the order of precedence for runtime settings.

Reply:
Something is missing here but the logic it's close. We basically need to implement the config priority here similar to `LogLevel` as Tim mentioned ; it should be something like:
* If the config is set by RC (`mergedConfig.EnableStreamLogs != nil`), it should be set accordingly
* If not but it was previously set by RC (`mergedConfig.EnableStreamLogs != nil && streamLogsSource == model.SourceRC`), revert `config.Datadog().UnsetForSource("enable_stream_logs", model.SourceRC)`
* Else, just do nothing and leave it as is

Also, I'm wondering if you should not register the `enable_stream_logs` key just like `log_level` and the other runtime settings (see [here](https://github.com/DataDog/datadog-agent/blob/201b789380d9c9faac6339d714aced6f0f4a45d8/pkg/config/settings/runtime_setting_log_level.go))